lib/c-api/src/wasm_c_api/instance.rs

```diff
@@ -7,11 +7,28 @@ use std::mem;
 use std::sync::Arc;
 use wasmer::{Extern, Instance, InstantiationError};
 
+/// Opaque type representing a WebAssembly instance.
 #[allow(non_camel_case_types)]
 pub struct wasm_instance_t {
     pub(crate) inner: Arc<Instance>,
 }
 
+/// Creates a new instance from a WebAssembly module and a
+/// set of imports.
+///
+/// ## Errors
+///
+/// The function can fail in 2 ways, as defined by the specification:
+///
+/// 1. Link errors that happen when plugging the imports into the
+///    instance,
+/// 2. Runtime errors that happen when running the module `start`
+///    function.
+///
```

Comment:
I think we should be clear about the fact that errors will not result in panics but in the `traps` being set.

Also, you are refering to the "specification": it could be a good thing to add a link to it don't you think?

Reply:
I don't find the paragraph in the spec anymore. I've removed that part of the sentence :-).